libsplinter/src/network/peer_manager/notification.rs

```diff
@@ -21,6 +21,10 @@ use std::sync::mpsc::{Receiver, TryRecvError};
 pub enum PeerManagerNotification {
     Connected { peer: String },
     Disconnected { peer: String },
+    PeerReady { peer: String },
+    // returned if the peer has failed for some reason, such as authorization failure and any
+    // peer refs should be dropped
+    BadPeer { peer: String },
```

Comment:
This should be a doc comment.  `BadPeer` isn't a great name, but I don't yet have better name.  We can fix rename that in followup PR.

Reply:
changed to doc comments, ya i don't love that name either